internal/repositories/repositories.go

```diff
@@ -99,6 +103,38 @@ func (r *Repository) GetSignalLatest(ctx context.Context, latestArgs *model.Late
 	return coll, nil
 }
 
+// GetDeviceActivity returns device status activity level.
+func (r *Repository) GetDeviceActivity(ctx context.Context, vehicleTokenID int, adManuf string) (*model.DeviceActivity, error) {
+	mtrLower := strings.ToLower(adManuf)
+	if mtrLower == hashDogMtr {
```

Comment:
I think we should lift this up to the level of the map from `"autopi"` to the source field.